server/task_executor.go

```diff
@@ -158,8 +161,29 @@ func runMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.
 		return true, nil, err
 	}
 
+	// For tenant mode project, we will only write back latest schema file on the last task.
+	tenantWriteBack := true
+	project, err := server.composeProjectByID(ctx, task.Database.ProjectID)
+	if err != nil {
+		return true, nil, err
+	}
+	if project.TenantMode == api.TenantModeTenant {
+		var lastTask *api.Task
+		for i := len(issue.Pipeline.StageList) - 1; i >= 0; i-- {
```

Comment:
will null exception if issue is nil. Our model doesn't enforce pipeline must belong to an issue.

Reply:
Done.

Reply:
issue != nil can stay after L165

Reply:
Done